### PR TITLE
Adding Test for DbInfoController and SchemaController  [resolves #715]

### LINKED ISSUE #715: Add integration tests for DB metadata API

### PR DESCRIPTION
**Pull request by Shrman007658**
This class is responsible for returning the metadata about the databases connected to the service. Added one test for the getObjects method which returns a list of DbInfoObject.
resolves #715

### CODE REVIEW

**Review by kdhrubo — CHANGES_REQUESTED** [review]
This should be integration test.

File: api-rest/src/test/java/com/homihq/db2rest/jdbc/rest/meta/schema/SchemaControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        actualTableObject = schemaController.getObjects("key","type==notPresentInList");
+        assertEquals(0,actualTableObject.size());
+    }
+    @Test
+    void testGetObjectsValidFilter_invalidFieldFilter() {
+        when(jdbcManager.getDbMetaByDbId("key")).thenReturn(metadataMap.get("key"));
+        List<TableObject> actualTableObject;
+        actualTableObject = schemaController.getObjects("key","invalidFilterField==notPresentInList");
+        assertEquals(0,actualTableObject.size());
```
Comment (thadguidry) [inline api-rest/src/test/java/com/homihq/db2rest/jdbc/rest/meta/schema/SchemaControllerTest.java:92]: Hmm, that looks awkward.  We filter against a field, it's not found, and so we return an empty list.
@kdhrubo Should we improve the code to return a simple message such as "No matches found."?  Or is empty List enough?  If empty List is assumed to be the same as "No matches found.", then we should change this test result to not check a size, but ensure what actually is being returned with `assertEquals(actualTableObject.isEmpty())`

Some say "always return some object or message, never null".  Others say its fine for APIs to return nothing.  But even nothing is always represented by some object or message.  In this case, an empty List object.  So we should improve this.

**Review by thadguidry — CHANGES_REQUESTED** [review]

**Comment by kdhrubo** [comment]
He has written unit test. As Lenur wrote in the issue we rely on
integration test.

**Comment by Shrman007658** [comment]
Okay, I will try to add Integration tests as well. But as a good to have are we keeping the Unit Tests at all or should I remove both of the classes?

File: api-rest/src/test/java/com/homihq/db2rest/jdbc/rest/meta/schema/SchemaControllerTest.java (hunk 1 above)
Comment (Shrman007658) [reply api-rest/src/test/java/com/homihq/db2rest/jdbc/rest/meta/schema/SchemaControllerTest.java:92]: For now I have added both IT and UT for the current code itself. 
Do we have a plan for when this change will be done?

**Comment by Shrman007658** [comment]
kindly requesting re review

**Review by kdhrubo — APPROVED** [review]

**Pull request merged**